### PR TITLE
Implementation of the backend template conversion

### PR DESCRIPTION
**Pull request by joshcampitelli**
### Testing the conversion of templates to `.js` files:
**Direct testing of `convertTemplate` function:** 
`cd backend`
`npm i`
`npm test`
**Note:** will generate file `test/generated-file.js`

**Indirect testing through the `express` server using _Postman_ 📪:**
`cd backend`
`npm i`
`node server/index.js`
```
POST to: http://localhost:3000/generate-file
BODY: 
{
    "template": "test-template.js",
    "name": "Josh",
    "hometown": "Ottawa, ON, Canada",
    "kids": [
        {
            "name": "Ahmed Sakr",
            "age": "22"
        },
        {
            "name": "Michael",
            "age": "23"
        }
    ]
}
```
**Note:** will also generate file `test/generated-file.js`

### CODE REVIEW

File: backend/server/index.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+        extended: true,
+    })
+)
+
+/**
+ * Pass the template name in request.headers.template
+ * Example to run the test file: "key: test-template.js"
+ */
+app.post('/generate-file', (request, response) => {
+    converter.convertTemplate(request.headers.template, request.body);
```
Comment (ahmedsakr) [inline backend/server/index.js:22]: why pass the template in the header? it's probably easier in the body

File: backend/template-generation/test/test-template-generation.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
+let converter = require('../generate-source-file.js');
+let data = {
```
Comment (ahmedsakr) [inline backend/template-generation/test/test-template-generation.js:2]: We will have to figure out precisely what data we want from LinkedIn

**Review by MichaelCalabro — APPROVED** [review]

File: backend/server/index.js (hunk 1 above)
Comment (joshcampitelli) [reply backend/server/index.js:22]: That's a good point, I've added the template to the JSON object as `data.template`

File: backend/template-generation/test/test-template-generation.js (hunk 2 above)
Comment (joshcampitelli) [reply backend/template-generation/test/test-template-generation.js:2]: This is just for testing the functionality of the script, once we have sample data from LinkedIn we can create tests for that as well.

**Review by ahmedsakr — APPROVED** [review]

**Pull request merged**